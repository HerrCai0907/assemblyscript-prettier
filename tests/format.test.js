import * as prettier from "prettier";
import asPlugin from "../src/plugin.js";

test("variant decorator", async () => {
  const originCode = `

@lazy let offset: usize = startOffset;

`;

  const formatedCode = await prettier.format(originCode, {
    parser: "typescript",
    plugins: [asPlugin],
  });

  expect(formatedCode).toMatchSnapshot();
});

test("function decorator", async () => {
  const originCode = `
  
    @global @unsafe
    export function __new(size: usize, id: i32): usize {}
    
  
  `;

  const formatedCode = await prettier.format(originCode, {
    parser: "typescript",
    plugins: [asPlugin],
  });

  expect(formatedCode).toMatchSnapshot();
});

test("class decorator", async () => {
  const originCode = `

      @global @unsafe
      export class AA

      {}

    `;

  const formatedCode = await prettier.format(originCode, {
    parser: "typescript",
    plugins: [asPlugin],
  });

  expect(formatedCode).toMatchSnapshot();
});
