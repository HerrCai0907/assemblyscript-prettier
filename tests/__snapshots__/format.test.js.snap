// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class decorator 1`] = `
"
@global @unsafe
export class AA {}
"
`;

exports[`function decorator 1`] = `
"
@global @unsafe
export function __new(size: usize, id: i32): usize {}
"
`;

exports[`variant decorator 1`] = `
"
@lazy let offset: usize = startOffset;
"
`;
